### DISCUSSION TITLE
Exists any way to get the readings from JSON command?

### DISCUSSION THREAD

**Discussion by miky-boy** [arendst/Tasmota#15416] [Q&A] [answered]
Hi guys,
I have the ZMAI-90 with the last tasmota version. I use the MQTT since a long time ago, but now i want also to provide this readings by a JSON query.
I already use the cmnd= status 0 for other things, but to get this readings exist any command or possible solution?
I already read all the commands list and didn't found any command.
I realized now exists a new version of this ZMAI-90 module with TuyaMCU, but i think my module is older and doesn't support the TuyaMCU, or i'm wrong?
Many thanks and Regards for all.

**Comment by sfromis** [comment]
Without examples, it is not clear what you specifically want. 

To the extent that Tasmota is publishing a JSON payload to MQTT, you can generally create triggers to match the specific object/fields names used in that JSON. If you have "something" publish data in JSON format, you can use the Tasmota [subscribe](https://tasmota.github.io/docs/Subscribe-&-Unsubscribe/) feature to pick single fields from that. You have better features for this with ESP32 and Berry.

**Comment by miky-boy** [comment, marked as answer]
Hi sfromis, thanks for your message.

Nowadays i have my device sending the readings for my raspberry pi thought MQTT service and works fine to create graph on influxdb and grafana. My goal now is to create a ESP32 touch screen to see all readings from my home with other esp8266 spread on my home to read temperature and humidity :), like a dashboard.

So the idea is to get the energy readings directly from the ZMAI-90 device with the tasmota firmware to avoid get the readings from the influxdb http. I remember this possibility because if i execute the command "http://192.168.2.201/cm?user=xxx&password=xxx&cmnd=status%206" i got some json output from the device, in this case with the mqtt settings ex: 

{"StatusMQT":{"MqttHost":"192.168.2.17","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_DE36E9","MqttUser":"MQTT_Server_Username","MqttCount":1,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}

I don't know if is possible to retrieve the readings from serial (in my case with this ZMAI version) directly by json output. The real output from the device is something like, where the SerialReceived xxx is the readings in hex and inverted. I have the login behind this long reading, what i want is to get it on JSON. 

17:24:33.959 MQT: tele/SensorQuadro/RESULT = {"SerialReceived":"FE01087750840007240000836402000050000008820500482202003938060013090000EE"}

Do you know if is possible, and how?

Many Thanks and Regards,

**Reply by sfromis** [reply]
It is still not clear what your specific scenario is, what data flow you envision, what you operationally mean by "directly by json output" vs "real output". You post that "SerialReceived" message, which is a JSON formatting published to MQTT. On an ESP32 device, you can retrieve that, and parse it, assuming that you know how it is built. Where are you expecting that "what I want is to get it on JSON" to take place?

**Reply by miky-boy** [reply]
Hi sfromis.

Nowadays the device send via MQTT the reading. It's fine and i will keep it for graphic purposes.

What i want now is the possibility to have other independent device ESP32 (in case) to create a query to the enegy device, like the example "http://192.168.2.201/cm?user=xxx&password=xxx&cmnd=status%206" and got the reading like the example "FE01087750840007240000836402000050000008820500482202003938060013090000EE"

Hope now sound better.
Thanks and Regards

**Reply by sfromis** [reply]
That does not indicate the data flow around that hex string.

**Reply by miky-boy** [reply]
Hi,
I'm not sure if I understood your question, but the flows are:

1) TASMOTA -> RPI -> Using MQTT ** This flow is working perfectly
2) ESP8266 -> TASMOTA -> USING broswerquery/JSON ** This is what i want to use, like if i put on my broswer the URL "http://192.168.2.201/cm?user=xxx&password=xxx&cmnd=status%206", where the IP "192.168.2.201" is the IP from my TASMOTA.

Thanks again and Regards

**Reply by barbudor** [reply]
As the data are already going through MQTT, just subscribe to the topic on your esp32 and the ESP32 will receive the same dat

**Reply by miky-boy** [reply]
Hi barbudor,

Well my goal is to remove the dependency from the RPI MQTT server to create an independent sensor system in my home to make it available when i leave this house. The sensor temperature and humidity already have two methods to provide the readings:
1) MQTT client configured
2) JSON information on request
The unique yet dependent from the MQTT server is the Energy device. 

I tried now to configure my device with the TuyaMCU template but i don't receive any reading. Probably because my device is too old and only works with the ZMAI-90 template. Because if the TuyaMCU receive readings this kind of information is present on the query "http://192.168.2.201/cm?user=xxx&password=xxx&cmnd=status%200".

I'm not sure if is possible to have this reading on the old template or what kind of other options could be used :(
Thanks and Regards

**Reply by sfromis** [reply]
Tasmota is not meant for providing data upon request, except for already existing commands. The general structure is that data is published to MQTT, and then it is up to the other end to subscribe and parse the data. You cannot easily create new responses in the style of Status commands, as this is already not how Tasmota is meant to be used.

The fact that the "M" in the Tasmota name is for MQTT represents a design choice for how it works best.

**Reply by miky-boy** [reply]
Hi sfromis,
Thanks for your message. Yes, i undersood and is not desired to put so much load on device, because he was designed to read in the best accurate mode.
I only tried to reuse the existing resources and after tested without sucess the last release on TuyaMCU mode i realized the readings appear on GUI and trought status command, so i was checking if is possible to have it too with the correct ZMAI-90 template on my device.
Thanks and Regards
